libsigner/src/v0/messages.rs

```diff
@@ -751,6 +751,80 @@ impl SignerMessageMetadata {
     }
 }
 
+/// The latest version of the block response data
+pub const BLOCK_RESPONSE_DATA_VERSION: u8 = 2;
+
+/// Versioned, backwards-compatible struct for block response data
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct BlockResponseData {
+    /// The version of the block response data
+    pub version: u8,
+    /// The block response data
+    pub tenure_extend_timestamp: u64,
+    /// The unknown block response data bytes
+    pub unknown_bytes: Vec<u8>,
+}
+
+impl BlockResponseData {
+    /// Create a new BlockResponseData for the provided tenure extend timestamp and unknown bytes
+    pub fn new(tenure_extend_timestamp: u64) -> Self {
+        Self {
+            version: BLOCK_RESPONSE_DATA_VERSION,
+            tenure_extend_timestamp,
+            unknown_bytes: vec![],
+        }
+    }
+
+    /// Create an empty BlockResponseData
+    pub fn empty() -> Self {
+        Self::new(u64::MAX)
+    }
+
+    /// Serialize the "inner" block response data. Used to determine the bytes length of the serialized block response data
+    fn inner_consensus_serialize<W: Write>(&self, fd: &mut W) -> Result<(), CodecError> {
+        write_next(fd, &self.tenure_extend_timestamp)?;
+        // write_next(fd, &self.unknown_bytes)?;
+        fd.write_all(&self.unknown_bytes)
+            .map_err(CodecError::WriteError)?;
+        Ok(())
+    }
+}
+
+impl StacksMessageCodec for BlockResponseData {
+    /// Serialize the block response data.
+    /// When creating a new version of the block response data, we are only ever
+    /// appending new bytes to the end of the struct. When serializing, we use
+    /// `bytes_len` to ensure that older versions of the code can read through the
+    /// end of the serialized bytes.
+    fn consensus_serialize<W: Write>(&self, fd: &mut W) -> Result<(), CodecError> {
+        write_next(fd, &self.version)?;
+        let mut inner_bytes = vec![];
+        self.inner_consensus_serialize(&mut inner_bytes)?;
+        let bytes_len = inner_bytes.len() as u32;
```

Comment:
`StacksMessageCodec` is implemented for `Vec<T: StacksMessageCodec>` already.  Please replace this:

```rust
let bytes_len = inner_bytes.len() as u32;
write_next(fd, &bytes_len)?;
fd.write_all(&inner_bytes).map_err(CodecError::WriteError)?;
```

with this:

```
write_next(fd, &inner_bytes)?;
```